### PR TITLE
Add examples under "Advanced usage" showing StyleGuide config options

### PR DESCRIPTION
**Pull request by nchammas**
This PR adds 2 brief examples showing basic options that the user can pass in when initializing a `StyleGuide`. For people automating pycodestyle tests I image these options are very commonly used, so they should be documented somewhere.

Currently, the only way I believe someone would be able to figure out how to use these options is to carefully eyeball [this code](https://github.com/PyCQA/pycodestyle/blob/eae54ff0e4c50ccc4507e95a2f8689fefb89e70e/pep8.py#L1815-L1822), which hides `StyleGuide`'s initialization parameters behind a `*args, **kwargs` pair.

This PR also makes some minor typo fixes. For reference, take a look at how the page looks today [on Read The Docs](http://pep8.readthedocs.org/en/latest/advanced.html).

### CODE REVIEW

**Comment by nchammas** [comment]
Is there anything I can do to get this reviewed and merged in? It's a relatively simple, doc-only change.

If this documents features that are technically not part of a public API, I don't mind closing this PR.

File: docs/advanced.rst [hunk 1]
```diff
@@ -46,7 +63,6 @@ Another example is related to the `feature request #143
 at the beginning and the end of a file.  This use case is easy to implement
 through a custom wrapper for the PEP 8 library::
 
-  #!python
```
Comment (sigmavirus24) [inline docs/advanced.rst]: Why did you remove this?

Comment (nchammas) [inline docs/advanced.rst]: It didn't seem to be required.

Looking at the [rendered doc](http://pep8.readthedocs.io/en/latest/advanced.html#skip-file-header), it appears that this is not being used to inform syntax highlighting, and is instead being displayed as-is, which is probably not intended.

The other blocks of code appear to be highlighted correctly without this snippet.

Comment (sigmavirus24) [inline docs/advanced.rst]: It's meant to indicate that this is an executable script. Regardless, the purpose of your PR was to *add* examples, not change existing ones.

```
#!
```

At the top of a file indicates that the script can be executed by the thing after `#!`, e.g., 

```
#!python
#!/usr/bin/env python
#!/usr/bin/env bash
#!/usr/bin/env perl
```

Comment (IanLee1521) [inline docs/advanced.rst]: For the time being I agree with @sigmavirus24 that this should remain. I made a new branch https://github.com/PyCQA/pycodestyle/compare/gh-500 that takes your commit and re-adds this line and updates the heading format for consistency.

I'm fine to go ahead and merge that if there are no other comments?

Comment (nchammas) [inline docs/advanced.rst]: Sure, go ahead.

For some reason I thought it was a broken syntax highlighting indicator, not a shebang, but I see now that I was mistaken. My PR fixed some other ReST formatting typos, which is what got me on that train of thought in the first place.

Comment (nchammas) [inline docs/advanced.rst]: > updates the heading format for consistency

I made "Configuring tests" a subsection of "Automated tests" intentionally, but I guess the whole "Advanced usage" guide is really about how to use pycodestyle in automated tests.
